src/tools/miri/tests/pass/float.rs

```diff
@@ -38,8 +38,8 @@ macro_rules! assert_approx_eq {
     }};
 
     ($a:expr, $b: expr) => {
-        // accept up to 64ULP (16ULP for host floats and 16ULP for miri artificial error and 32 for any rounding errors)
-        assert_approx_eq!($a, $b, 64);
+        // accept up to 32LP (4ULP for miri artificial error and 28ULP for any rounding errors)
+        assert_approx_eq!($a, $b, 32);
```

Comment:
Do we need 32, or do lower numbers also work?

Reply:
I'm a bit surprised 16 doesn't work, given that it should only be 4+4 (from host and Miri) and then maybe a factor of 2 if we cross to a different exponent. But okay, it is what it is again.

```suggestion
        // Miri only adds 4 ULP of error, but we're seeing up to 28 ULP in these tests for unknown reasons.
        // Let's just accept up to 32.
        assert_approx_eq!($a, $b, 32);
```

Reply:
My memory is vague, so take it with a grain of salt. I'll test it again.

Reply:
On my machine using 24ULP works for all the tests. During the first pr we increased this value to 64ULP because some tests did multiple operations which increased the ULP error, this is still the case, but on a much smaller case (we used 16ULP errors back then).

However, there are some operations (`powi`, `log10`) that introduce a bigger ULP error than 4 on both single and double precision. I have no explanation for why this happens.
```rust
// some examples of these failed tests
assert_approx_eq!(25f32.powi(-2), 0.0016f32);  // 6ULP, fails 6 runs
assert_approx_eq!(1f32.exp(), f32::consts::E); // 5ULP, fails 6 runs
assert_approx_eq!(1f64.exp(), f64::consts::E); // 5ULP, fails 6 runs
assert_approx_eq!(f64::consts::E.log10(), f64::consts::LOG10_E); // 7ULP, fails sometimes

```

Fast and algebraic operations are still introducing 16ULP errors, so these automatically fail the tests if we go below 16ULP. If I temporarily set these to 4ULP, all the tests pass using 7ULP as the accepted ULP error. I know this pr doesn't focus on these operations but this gives a clearer picture.

Should I stick to 24ULP or investigate further and try to alter accepted ULPs on some specific tests that use multiple operations and go lower than 24ULP. 

In a future pr we can set this value lower when changing the errors in `foreign_items` and the fast and algebraic operations.

Reply:
> Fast and algebraic operations are still introducing 16ULP errors, so these automatically fail the tests if we go below 16ULP.

Ah, of course. I'll sync latest Miri into rustc and then I'll ask you to rebase -- I have changed the error to 12 ULP there (4 for the host, 4 for Miri, 4 for extra error due to crossing to a different exponent). It seems like that will also work with your patch then, so you can just leave this unchanged. (There'll be a conflict when the sync lands.)